app/javascript/components/app.jsx

```
@@ -2,78 +2,26 @@ import 'regenerator-runtime/runtime';
 import { persistCache } from 'apollo-cache-persist';
 import WidgetController from '@components/widget-controller';
 import versionCompare from '@components/version-compare';
 import helioSchema from '@javascript/schema.json';
-import Cookies from 'js-cookie';
 import GlobalStyle from '../styles';
+import absintheLink from '../lib/phoenix-graphql-backend';
+import railsLink from '../lib/rails-graphql-backend';
 
-const cable = ActionCable.createConsumer(
-  `${process.env.RAILS_BACKEND_URI}/cable`,
-);
-
-const phoenixSocket = new PhoenixSocket(`${process.env.WEBSOCKET_URI}/socket`);
-
-const absintheSocket = AbsintheSocket.create(phoenixSocket);
-
-const webSocketLinkAbsinthe = createAbsintheSocketLink(absintheSocket);
-
-const webSocketLinkRails = new ActionCableLink({ cable });
-
-const httpLink = new HttpLink({
-  uri: `${
-    process.env.BACKEND_LANGUAGE === 'ruby'
-      ? process.env.RAILS_BACKEND_URI
-      : process.env.PHOENIX_BACKEND_URI
-  }/graphql`,
-  credentials: 'include',
-});
-
-const authLink = setContext((_, { headers }) => {
-  // Get the authentication token from the cookie if it exists.
-  const token = Cookies.get('token');
+let link = null;
 
-  // Add the new Authorization header.
-  return {
-    headers: {
-      ...headers,
-      authorization: token ? `Bearer ${token}` : '',
-    },
-  };
-});
-
-const authedHttpLink = authLink.concat(httpLink);
-
-const railsLink = ApolloLink.split(
-  ({ query }) => {
-    const { kind, operation } = getMainDefinition(query);
-    return kind === 'OperationDefinition' && operation === 'subscription';
-  },
-  webSocketLinkRails,
-  httpLink,
-);
-
-const absintheLink = split(
-  operation => hasSubscription(operation.query),
-  webSocketLinkAbsinthe,
-  authedHttpLink,
-);
+if (process.env.BACKEND_LANGUAGE === 'ruby') {
+  link = railsLink;
+} else if (process.env.BACKEND_LANGUAGE === 'elixir') {
+  link = absintheLink;
+}
```

Should we else here and fail loudly? Or just default to `ruby` for now?

---

I was thinking we default to ruby for now.